### ISSUE TITLE
Emulating game

### ISSUE THREAD

**Issue by BeefEX** [bengsfort/tile-pusher#5]
I just got a pretty interesting idea. Since this is done in Unity, it wouldn't be hard at all to add "Play mode" in which you could try the level with few common control schemes or character physics. With a possibility to tweak the physics, it would allow users to easily test their levels in the editor. Maybe it could be even possible to add custom controls and physics, not sure how that could work though.

**Comment by bengsfort** [comment]
Absolutely!~ My initial plan for this since this exact issue has been a pretty big one for me in the past was to include a "Player Reference" that can be toggled on/off and moved around the map, but this may be something we can add in addition to the references. For example:

-> Create the player _normal_  reference and add values for: player height (ie 2 tile), max jump height (ie 2.5 tiles), max jump distance (ie 4 tiles)
-> A reference player gets added on a layer above the map that can be dragged around; with lines showing the movement properties so the level designer can have an at-a-glance reference for spacing; preventing unnecessary map exports / game builds for trivial spacing tests.
-> Designer finishes initial spacing, enables test mode and the reference turns into a controllable, basic player with movement mimicking the properties added to the reference.

This could be expanded to have different types of references, like platformer references, dungeon crawler-type references, etc. with different movement visualization.

I'll attach a quick sketch I made for the referencing flow on this ticket so the features can stay together.
![playerreferences](https://cloud.githubusercontent.com/assets/6495586/26486900/bdb7c29c-4205-11e7-9a44-a0bb8f5a81b2.png)

**Comment by BeefEX** [comment]
That could definitely be combined, each idea is a bit different.

**Comment by bengsfort** [comment]
Yeah; but I think the foundations of each could probably be shared that way the designer doesn't have to keep inputting the same physics properties in multiple places

**Comment by BeefEX** [comment]
That would be nice.
